### PR TITLE
FEAT-#5620: Synchronize parameters of `apply_full_axis` with `broadcast_apply_full_axis`

### LINKED ISSUE #5620: Synchronize `apply_full_axis` and `broadcast_apply_full_axis` parameters
The following parameters make sense for both functions:
* apply_indices
* enumerate_partitions

After these changes, only one difference should remain: `other` parameter.

### PR DESCRIPTION
**Pull request by shalearkane**
## What do these changes do?

Add parameters 
- apply_indices
- enumerate_partitions
to `apply_full_axis` function in `modin/core/dataframe/pandas/dataframe/dataframe.py` to match with the parameters of the function `broadcast_apply_full_axis`.

- [x] first commit message and PR title follow format outlined [here]
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5620 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by shalearkane** [comment]
Do I have to add tests for the above changes to make sure this change is working?

**Comment by YarShev** [comment]
@shalearkane, please replace the use of `broadcast_apply_full_axis` to `apply_full_axis` in the places where we pass in `other=None`, i.e. in these places.

https://github.com/modin-project/modin/blob/a4504a0e298f6036dc9925bdee9a6898f4d64140/modin/core/storage_formats/pandas/query_compiler.py#L565

https://github.com/modin-project/modin/blob/a4504a0e298f6036dc9925bdee9a6898f4d64140/modin/experimental/core/execution/ray/implementations/pandas_on_ray/io/io.py#L241

https://github.com/modin-project/modin/blob/a4504a0e298f6036dc9925bdee9a6898f4d64140/modin/experimental/core/execution/unidist/implementations/pandas_on_unidist/io/io.py#L246

So after that you do not need to add tests because we already check it.

File: modin/core/storage_formats/pandas/query_compiler.py [hunk 1]
```diff
@@ -562,7 +562,7 @@ def _reset(df, *axis_lengths, partition_idx):
                 new_columns = None
 
             return self.__constructor__(
-                self._modin_frame.broadcast_apply_full_axis(
+                self._modin_frame.apply_full_axis(
                     axis=1,
                     func=_reset,
                     other=None,
```
Comment (YarShev) [inline modin/core/storage_formats/pandas/query_compiler.py:568]:
```suggestion
```
`apply_full_axis` method doesn't have `other` parameter so we should remove it when calling.

File: modin/experimental/core/execution/ray/implementations/pandas_on_ray/io/io.py [hunk 2]
```diff
@@ -238,7 +238,7 @@ def func(df, **kw):
             df.to_pickle(**kwargs)
             return pandas.DataFrame()
 
-        result = qc._modin_frame.broadcast_apply_full_axis(
+        result = qc._modin_frame.apply_full_axis(
             1, func, other=None, new_index=[], new_columns=[], enumerate_partitions=True
```
Comment (YarShev) [inline modin/experimental/core/execution/ray/implementations/pandas_on_ray/io/io.py:242]:
```suggestion
            1, func, new_index=[], new_columns=[], enumerate_partitions=True
```

File: modin/experimental/core/execution/unidist/implementations/pandas_on_unidist/io/io.py [hunk 3]
```diff
@@ -243,7 +243,7 @@ def func(df, **kw):
             df.to_pickle(**kwargs)
             return pandas.DataFrame()
 
-        result = qc._modin_frame.broadcast_apply_full_axis(
+        result = qc._modin_frame.apply_full_axis(
             1, func, other=None, new_index=[], new_columns=[], enumerate_partitions=True
```
Comment (YarShev) [inline modin/experimental/core/execution/unidist/implementations/pandas_on_unidist/io/io.py:247]:
```suggestion
            1, func, new_index=[], new_columns=[], enumerate_partitions=True
```

**Comment by YarShev** [comment]
@shalearkane, I applied some changes so CI should pass now.

**Review by YarShev — APPROVED** [review]
@shalearkane, thanks for the changes! LGTM.

**Pull request merged**